parquet-column/src/main/java/org/apache/parquet/io/api/Binary.java

```diff
@@ -148,7 +174,7 @@ public void writeTo(DataOutput out) throws IOException {
 
   private static class FromStringBinary extends ByteArrayBackedBinary {
     public FromStringBinary(byte[] value) {
-      super(value);
+      super(value, true);
```

Comment:
The conversion should be done in a static method. Then you can add exception handling to that method but still use it in the call to super.